File: demo-service-impl/src/main/java/com/github/zubmike/service/demo/dao/db/ZoneDaoImpl.java
package com.github.zubmike.service.demo.dao.db;

import org.hibernate.SessionFactory;
import com.github.zubmike.service.dao.db.BasicEntityItemDao;
import com.github.zubmike.service.demo.dao.ZoneDao;
import com.github.zubmike.service.demo.types.Zone;

import javax.inject.Inject;

public class ZoneDaoImpl extends BasicEntityItemDao<Integer, Zone> implements ZoneDao {

	@Inject
	public ZoneDaoImpl(SessionFactory sessionFactory) {
		super(sessionFactory, Zone.class);
	}

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/dao/db/ZoneDaoImpl.kt
package com.github.zubmike.service.demo.dao.db

import org.hibernate.SessionFactory
import com.github.zubmike.service.dao.db.BasicEntityItemDao
import com.github.zubmike.service.demo.dao.ZoneDao
import com.github.zubmike.service.demo.types.Zone

class ZoneDaoImpl (sessionFactory: SessionFactory) : BasicEntityItemDao<Int, Zone>(sessionFactory, Zone::class.java), ZoneDao
